redis/health.go

```diff
@@ -17,7 +20,17 @@ func NewHealthCheck(client redis.UniversalClient, name string) *HealthCheck {
 }
 
 func (r *HealthCheck) HealthChecks() (name string, ready, live func(ctx context.Context) error) {
-	ready = func(ctx context.Context) error {
+	ready = func(ctx context.Context) (err error) {
+		start := time.Now()
+		m := o11y.FromContext(ctx).MetricsProvider()
+
+		defer func() {
+			t := time.Since(start)
+			_ = m.TimeInMilliseconds("redis_healthcheck_duration", float64(t.Milliseconds()), []string{
+				fmt.Sprintf("%s:%t", "error", err != nil),
```

Comment:
Probably want a few more tags on this

Reply:
I added "name" to the tags. I'm not sure what else is really useful here. I might need to look at some other redis client-y metrics and see what tags they use